python/pyarrow/tests/test_pandas.py

```diff
@@ -221,6 +221,16 @@ def test_column_index_names_with_tz(self):
         )
         _check_pandas_roundtrip(df, preserve_index=True)
 
+    def test_column_index_names_with_decimal(self):
```

Comment:
Could you add info about the GitHub issue this test is added for? Similarly as what you added in  `reconstruct_columns_from_metadata`  or in the test just above.

Reply:
@AlenkaF sure, done.